src/main/java/com/iota/iri/service/snapshot/conditions/SnapshotDepthCondition.java

```diff
@@ -0,0 +1,78 @@
+package com.iota.iri.service.snapshot.conditions;
+
+import com.iota.iri.conf.SnapshotConfig;
+import com.iota.iri.service.snapshot.Snapshot;
+import com.iota.iri.service.snapshot.SnapshotException;
+import com.iota.iri.service.snapshot.SnapshotProvider;
+
+import com.google.common.annotations.VisibleForTesting;
+
+/**
+ * 
+ * Conditions for snapshotting based on the distance between the current milestone and the amount we want to keep 
+ *
+ */
+public class SnapshotDepthCondition implements SnapshotCondition {
+
+    private final SnapshotConfig config;
+    private final SnapshotProvider snapshotProvider;
+
+    /**
+     * Implements a {@link SnapshotCondition} based on the amount of milestones we have
+     *
+     * @param config configuration with snapshot specific settings.
+     * @param snapshotProvider gives us access to the relevant snapshots.
+     */
+    public SnapshotDepthCondition(SnapshotConfig config, SnapshotProvider snapshotProvider) {
+        this.config = config;
+        this.snapshotProvider = snapshotProvider;
+    }
+
+    @Override
+    public boolean shouldTakeSnapshot(boolean isInSync) {
+        int localSnapshotInterval = getSnapshotInterval(isInSync);
+
+        int latestSnapshotIndex = snapshotProvider.getLatestSnapshot().getIndex();
+        int initialSnapshotIndex = snapshotProvider.getInitialSnapshot().getIndex();
+
+        return latestSnapshotIndex - initialSnapshotIndex > config.getLocalSnapshotsDepth() + localSnapshotInterval;
+    }
+
+    /**
+     * <p>
+     * This method determines the milestone that shall be used for the local snapshot.
+     * </p>
+     * <p>
+     * It determines the milestone by subtracting the {@link SnapshotConfig#getLocalSnapshotsDepth()} from the latest
+     * solid milestone index and retrieving the next milestone before this point.
+     * </p>
+     * 
+     * @param tangle Tangle object which acts as a database interface
+     * @param snapshotProvider data provider for the {@link Snapshot}s that are relevant for the node
+     * @param config important snapshot related configuration parameters
```

Comment:
Where are the paramerers?
Comment docs are also code we have to maintatin unfortunately

Reply:
Not sure this is fixed, can you double check

Reply:
its gone